yt/visualization/_commons.py

```diff
@@ -96,3 +96,55 @@ def get_canvas(figure, filename):
             f"without an extension."
         )
     return get_canvas_class(suffix)(figure)
+
+
+T = TypeVar("T", tuple, list)
+
+
+def _swap_axes_extents(extent: T) -> T:
+    """
+    swaps the x and y extent values, preserving type of extent
+
+    Parameters
+    ----------
+    extent : sequence of four unyt quantities
+        the current 4-element tuple or list of unyt quantities describing the
+        plot extent. extent = (xmin, xmax, ymin, ymax).
+
+    Returns
+    -------
+    tuple or list
+        the extent axes swapped, now with (ymin, ymax, xmin, xmax).
+
+    """
+    extent_swapped = [extent[2], extent[3], extent[0], extent[1]]
+    return type(extent)(extent_swapped)
+
+
+def _swap_arg_pair_order(*args):
+    """
+    flips adjacent argument pairs, useful for swapping x-y plot arguments
+
+    Parameters
+    ----------
+    *args
+        argument pairs, must have an even number of *args
+
+    Returns
+    -------
+    tuple
+        args  with order of pairs switched, i.e,:
+
+        _swap_arg_pair_order(x, y, px, py) returns:
+            y, x, py, px
+
+    """
+
+    assert len(args) % 2 == 0
+    n_pairs = int(len(args) / 2)
```

Comment:
use euclidian division directly
```suggestion
    n_pairs = len(args) // 2
```